pkg/querier/queryrange/codec.go

```diff
@@ -231,10 +231,8 @@ func (r *LabelRequest) GetStep() int64 {
 
 func (r *LabelRequest) WithStartEnd(s, e time.Time) queryrangebase.Request {
 	clone := *r
-	tmp := s
-	clone.Start = &tmp
-	tmp = e
-	clone.End = &tmp
+	clone.Start = &s
```

Comment:
previous code was incorrect since it writes to tmp twice, which is referred by both start and end

I could add two new local vars and take pointer to them, but since it is pass by value I though we could directly point to the function arguments

Reply:
Oh you're completely right, nice 👍